src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,196 @@
-import { Component } from 'react';
+import classnames from 'classnames';
+import React, { useState } from 'react';
+import './NewMovie.scss';
 
-type Props = {};
-type State = {};
+type Props = {
+  onAdd: (movie: Movie) => void;
+  movies: Movie[]
+};
 
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
+export const NewMovie: React.FC<Props> = ({ onAdd, movies }) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [titleError, setTitleError] = useState('');
+  const [imdbIdError, setImdbIdError] = useState('');
+  const [imgUrlError, setImgUrlError] = useState('');
+  const [imdbUrlError, setImdbUrlError] = useState('');
+  const checkRepeatedId = movies.some(movie => movie.imdbId === imdbId);
+  // eslint-disable-next-line max-len
+  const isButtonDisabled = !title || !imdbId || !imgUrl || !imdbUrl || checkRepeatedId;
 
-  render() {
-    return (
-      <form>
-        Put the form here
+  const errorPhrase = (atr: string) => {
+    return `Enter ${atr}, please`;
+  };
+
+  const handleSubmit = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+  };
+
+  const validateField = (
+    atr: string,
+    func: (value: React.SetStateAction<string>) => void,
+    atrName: string,
+  ) => {
+    func('');
+
+    if (!atr) {
+      func(errorPhrase(atrName));
+    }
+  };
+
+  const reset = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const handleChange = () => {
+    if (title && imdbId.length === 9 && imgUrl && imdbUrl && !checkRepeatedId) {
+      onAdd({
+        title,
+        description,
+        imdbId,
+        imgUrl,
+        imdbUrl,
+      });
+
+      reset();
+    }
+  };
+
+  const controlValidId = (str: string) => {
+    return str.split('').map((char, i) => {
+      if (i < 2) {
+        return 't';
+      }
+
+      return char;
+    }).filter((_, i) => i < 9).join('');
+  };
+
+  return (
+    <>
+      <form onSubmit={handleSubmit} className="form">
+        <div className="form__unit">
+          <input
+            type="text"
+            name="title"
+            placeholder="title"
+            className={classnames('input form__input', {
+              'is-danger': !title && titleError,
```

Comment:
titleError means that there is no title, whatsoever? fix it everywhere
```suggestion
              'is-danger': titleError,
```